ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/ccmv2/GrpcCcmV2Client.java

```diff
@@ -51,17 +53,20 @@ public InvertingProxy getOrCreateInvertingProxy(String requestId, String account
         }
     }
 
-    public InvertingProxyAgent registerAgent(String requestId, String accountId, String domainName, String keyId, String actorCrn) {
+    public InvertingProxyAgent registerAgent(String requestId, String accountId, Optional<String> environmentCrnOpt, String domainName,
+        String keyId, String actorCrn) {
         try (ManagedChannelWrapper channelWrapper = makeWrapper()) {
             ClusterConnectivityManagementV2BlockingStub client = makeClient(channelWrapper.getChannel(), requestId, actorCrn);
-            RegisterAgentRequest registerAgentRequest =  RegisterAgentRequest.newBuilder()
-                    .setDomainName(domainName)
+            RegisterAgentRequest.Builder registerAgentRequestBuilder =  RegisterAgentRequest.newBuilder()
                     .setAccountId(accountId)
-                    .setKeyId(keyId)
-                    .build();
-
-            LOGGER.debug("Calling registerAgent with params accountId: '{}', domainName: '{}', keyId:'{}' ",
-                    accountId, domainName, keyId);
+                    .setDomainName(domainName)
+                    .setKeyId(keyId);
+            if (environmentCrnOpt.isPresent()) {
+                registerAgentRequestBuilder.setEnvironmentCrn(environmentCrnOpt.get());
+            }
```

Comment:
just a nit: could be simplified using ifPresent, which looks like:
```
environmentCrnOpt.ifPresent(envCrn -> registerAgentRequestBuilder.setEnvironmentCrn(envCrn));

Reply:
done.